test/Passes/tile-and-fuse.mlir

```diff
@@ -273,3 +273,424 @@ func.func @matmul_sequence_fusion(%arg0: tensor<32x32xf32>, %arg1: tensor<32x32x
+  // CHECK: %{{.+}} = linalg.generic
+  // CHECK-SAME:  indexing_maps = [#[[MAP]], #[[MAP1]], #[[MAP2]]]
+  // CHECK-SAME:  iterator_types = ["reduction", "parallel", "parallel", "reduction"]
+  // CHECK:       ^bb0(
+  // CHECK-NEXT:    arith.mulf
+  // CHECK-NEXT:    arith.addf
+  // CHECK: %{{.+}} = linalg.generic
+  // CHECK-SAME:  indexing_maps = [#[[MAP3]], #[[MAP4]], #[[MAP3]]]
+  // CHECK-SAME:  iterator_types = ["parallel", "parallel"]
+  // CHECK:       ^bb0(
+  // CHECK-NEXT:    arith.addf
+  // CHECK: linalg.generic
+  // CHECK-SAME:  indexing_maps = [#[[MAP3]], #[[MAP3]]]
+  // CHECK-SAME:  iterator_types = ["parallel", "parallel"]
+  // CHECK:       ^bb0(
+  // CHECK-NEXT:    arith.maxf
+  // CHECK: }
+  // CHECK: }
+  %36 = linalg.generic {indexing_maps = [#map, #map1, #map2], iterator_types = ["parallel", "parallel", "reduction", "parallel", "parallel", "reduction"]} ins(%35, %arg28 : tensor<8x112x32x32xbf16>, tensor<112x112x32x32xbf16>) outs(%arg30 : tensor<8x112x32x32xbf16>) {
+    ^bb0(%in: bf16, %in_0: bf16, %out: bf16):
+      %mul = arith.mulf %in, %in_0 : bf16
+      %add = arith.addf %out, %mul : bf16
+      linalg.yield %add : bf16
+  } -> tensor<8x112x32x32xbf16>
+  %expanded10 = tensor.expand_shape %arg29 [[0, 1]] : tensor<3584xbf16> into tensor<112x32xbf16>
+  %37 = tensor.empty() : tensor<8x112x32x32xbf16>
+  %38 = linalg.generic {indexing_maps = [#map3, #map4, #map3], iterator_types = ["parallel", "parallel", "parallel", "parallel"]} ins(%36, %expanded10 : tensor<8x112x32x32xbf16>, tensor<112x32xbf16>) outs(%37 : tensor<8x112x32x32xbf16>) {
+    ^bb0(%in: bf16, %in_0: bf16, %out: bf16):
+      %add = arith.addf %in, %in_0 : bf16
+      linalg.yield %add : bf16
+  } -> tensor<8x112x32x32xbf16>
+  %39 = linalg.generic {indexing_maps = [#map3, #map3], iterator_types = ["parallel", "parallel", "parallel", "parallel"]} ins(%38 : tensor<8x112x32x32xbf16>) outs(%37 : tensor<8x112x32x32xbf16>) {
+    ^bb0(%in: bf16, %out: bf16):
+      %max = arith.maxf %in, %cst : bf16
+      linalg.yield %max : bf16
+  } -> tensor<8x112x32x32xbf16>
+  // CHECK: scf.for %{{.+}} = %[[C0]] to %[[C8]] step %[[C1]]
+  // CHECK-NEXT: scf.for %{{.+}} = %[[C0]] to %[[C112]] step %[[C1]]
+  // CHECK: %{{.+}} = linalg.generic
+  // CHECK-SAME:  indexing_maps = [#[[MAP]], #[[MAP1]], #[[MAP2]]]
+  // CHECK-SAME:  iterator_types = ["reduction", "parallel", "parallel", "reduction"]
+  // CHECK:       ^bb0(
+  // CHECK-NEXT:    arith.mulf
+  // CHECK-NEXT:    arith.addf
+  // CHECK: %{{.+}} = linalg.generic
+  // CHECK-SAME:  indexing_maps = [#[[MAP3]], #[[MAP4]], #[[MAP3]]]
+  // CHECK-SAME:  iterator_types = ["parallel", "parallel"]
+  // CHECK:       ^bb0(
+  // CHECK-NEXT:    arith.addf
+  // CHECK: linalg.generic
+  // CHECK-SAME:  indexing_maps = [#[[MAP3]], #[[MAP3]]]
+  // CHECK-SAME:  iterator_types = ["parallel", "parallel"]
+  // CHECK:       ^bb0(
+  // CHECK-NEXT:    arith.maxf
+  // CHECK: }
+  // CHECK: }
+  %c0 = arith.constant 0: index
```

Comment:
This part can go, all it does is print 4x4 slice of the result. I added it to verify results manually in my benchmark.

Reply:
Actually, let's keep it. The last relu was not fusing because it has multiple users (the extract_slice and the return). So in part the this last step exercise what we want to test.